addons/web/controllers/main.py

```diff
@@ -1956,6 +1956,10 @@ def report_routes(self, reportname, docids=None, converter=None, **data):
             if data['context'].get('lang'):
                 del data['context']['lang']
             context.update(data['context'])
+        else:
+            cids = request.httprequest.cookies.get('cids', str(request.env.user.company_id.id))
```

Comment:
```suggestion
            cids = request.httprequest.cookies.get('cids', str(request.env.company.id))
```
🤔

Reply:
no, that's actually the whole point of this PR... `request.env.company` is incorrect because `allowed_company_ids` is incorrect.

Reply:
ah, ok